models/product.js

```diff
@@ -0,0 +1,92 @@
+const mongoose = require("mongoose");
+const Joi = require("joi");
+Joi.objectId = require("joi-objectid")(Joi);
+
+const productSchema = new mongoose.Schema({
+  title: { type: String, maxLength: 50, default: "" },
+  brand: { type: String, default: "" },
+  averageRating: { type: Number, default: 0 },
+  totalRating: { type: Number, default: 0 },
+  discountPercentage: { type: Number },
+  actualPrice: { type: Number },
+  sellingPrice: { type: Number },
+  status: { type: String, enum: ["active", "inactive"], default: "active" },
+  subImages: { type: Array },
+  image: { type: String },
+  description: { type: String, default: "" },
+  totalSold: { type: Number, default: 0 },
+  totalInWishlist: { type: Number, default: 0 },
+  category: { type: Array },
+  creationDate: {
+    type: Date,
+    default: () => {
+      return new Date();
+    },
+  },
+  insertDate: {
+    type: Number,
+    default: () => {
+      return Math.round(new Date() / 1000);
```

Comment:
Why not use a `Date` methods as opposed to doing this rounding? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date